lib/protobuf/new_transform.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Protobuf.NewTransform do
+  @moduledoc """
+  For structs with a `Protobuf.TransformModule` set, the `c:Protobuf.new/1` function does not get called.
+  This transform module brings back the original behavior of calling `c:Protobuf.new/1` before the value
+  is encoded.
+  """
+
+  @behaviour Protobuf.TransformModule
+
+  @impl true
+  def encode(value, type), do: type.new!(value)
```

Comment:
I think `new!/1` is not backwards compatible. The previous behavior was `type.new(value)`, no?